### PR TITLE
[docs] Revamp storage migration

### PR DESCRIPTION
**Pull request by daenney**
# Description

* Some slightly nicer sections
* More links to authoritative documentation on the AWS side
* Mention the avatars and headers thing, in relation to #2328

## Checklist

Please put an x inside each checkbox to indicate that you've read and followed it: `[ ]` -> `[x]`

If this is a documentation change, only the first checkbox must be filled (you can delete the others if you want).

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [ ] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [ ] I/we have performed a self-review of added code.
- [ ] I/we have written code that is legible and maintainable by others.
- [ ] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have made any necessary changes to documentation.
- [ ] I/we have added tests that cover new code.
- [ ] I/we have run tests and they pass locally with the changes.
- [ ] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

**Comment by daenney** [comment]
https://gotosocial--2571.org.readthedocs.build/en/2571/configuration/storage/

**Comment by NyaaaWhatsUpDoc** [comment]
i'll leave review to @tsmethurst here as i'm not much of an s3 user myself :innocent:

File: docs/configuration/storage.md [hunk 1]
```diff
@@ -69,78 +69,88 @@ storage-s3-secret-key: ""
...
+    * `storage-s3-bucket` -> The `<bucketname>` that you created just now
+
+## Storage migration
+
+Migration between backends is freely possible. To do so, you only have to move the directories (and their contents) between the different implementations.
+
+When moving from one backend to another, the database will still contain references to headers and avatars from remote accounts pointing to the old storage backend which may result in them not loading correctly in clients. This will resolve itself over time, but you can force GoToSocial to refetch the avatar and header the next time you interact with a remote account by running the following SQL query on your database:
+
+```sql
+UPDATE accounts SET (avatar_media_attachment_id, avatar_remote_url, header_media_attachment_id, header_remote_url, fetched_at) = (null, null, null, null, null) WHERE domain IS NOT null;
```
Comment (tsmethurst) [inline docs/configuration/storage.md:120]: I'd mention here as well that you should restart your instance after running the query to clear the caches out. Or run the query while the instance is down.

**Review by tsmethurst — COMMENTED** [review]
Just one comment :) Feel free to squerge after that's resolved.

**Pull request merged**